lib/facility.rb

```diff
@@ -1,14 +1,63 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
-  end
-
-  def add_services(service)
-    @services << service
-  end
-end
+
+    attr_reader :name,
+                :address, 
+                :phone, 
+                :services,
+                :registered_vehicles,
+                :collected_fees
+
+    def initialize(facility_info)
+        @name = facility_info[:name] 
+        @address = facility_info[:address]
+        @phone = facility_info[:phone]
+        @services = []
+        @registered_vehicles = []
+        @collected_fees = 0
+    end
+
+    # Add Service
+    def add_service(service)
+        @services << service
+    end
+
+    # Register a Vehicle
+    def register_vehicle(vehicle)
+        if @services.include? ("Vehicle Registration")
+            vehicle.change_registration_date
+            vehicle.update_plate_type
+            @collected_fees += vehicle.fees
+            @registered_vehicles << vehicle
+        else
+            return nil
+        end
+    end
+
+    # Administer a Written Test
+    def administer_written_test(registrant)
+        if @services.include? ("Written Test")
+            registrant.are_written_eligible?
+        else
+            return false
+        end
+    end
+
+    # Administer a Road Test
+    def administer_road_test(registrant)
+        if @services.include? ("Road Test")
+            registrant.are_road_eligible?
+        else
+            return false
+        end
```

Comment:
This is good. Again, I like how you created a helper method in the Registrant class. You could refactor this even more by removing the if statement completely and simply having this in your method:
`@services.include?("Road Test") && registrant.are_road_eligble?` Just having that would return true or false for you.